src/jit/utils.cpp

```diff
@@ -1954,6 +1954,13 @@ float FloatingPointUtils::round(float x)
     return _copysignf(flrTempVal, x);
 }
 
+bool FloatingPointUtils::isPow2(double x)
+{
+    int exp;
+    double f = frexp(x, &exp);
+    return abs(f) == 0.5 && x != 0 && isnormal(1.0 / x);
```

Comment:
This is making a lot of assumptions about the crt implementation being correct/IEEE compliant and the floating-point state being in the correct format...

I much preferred the more explicit implementation.

Reply:
I wonder if I can use `> 0 && <= 0.5` instead of `== 0.5`

Reply:
I don't think that will be correct, since it would then cover non-powers of two.

The best behavior, IMO, would be to continue checking the raw bits (which is also notably what Clang is doing here) and not rely on the built-in CRT library.

Reply:
@tannergooding do you have a link to llvm piece of code for it? 🙂 I tried to find it in llvm and gcc codebases but with no luck 🙁

Reply:
The relevant bits are here:
* https://github.com/llvm-mirror/llvm/blob/master/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp#L1112
* https://github.com/llvm-mirror/llvm/blob/master/lib/IR/Constants.cpp#L227
* https://github.com/llvm-mirror/llvm/blob/master/lib/Support/APFloat.cpp#L3614

Looks like we already have the relevant 3rd party notice for LLVM: https://github.com/dotnet/coreclr/blob/master/THIRD-PARTY-NOTICES.TXT#L109

Reply:
> gcc codebases

Watch out with that, it's GPL code...

Reply:
Ok, thanks! Rolled back to bit twiddling impl (it's not final - I am now investigating the corner cases (denormals, etc)).